### ISSUE TITLE
Error creating containers - a container has already joined the endpoint

### ISSUE THREAD

**Issue by tomdee** [docker/docker#14107]
Simple repro.

The setup
```
curl -sSL https://experimental.docker.com/ | sudo sh
...
$ docker version 
Client version: 1.8.0-dev
Client API version: 1.20
Go version (client): go1.4.2
Git commit (client): f39b9a0
OS/Arch (client): linux/amd64
Experimental (client): true
Server version: 1.8.0-dev
Server API version: 1.20
Go version (server): go1.4.2
Git commit (server): f39b9a0
OS/Arch (server): linux/amd64
Experimental (server): true
```

The repro
```
docker rm -f delete_me; while  docker run -tid --net=host --name delete_me busybox; do docker rm -f delete_me; done
delete_me
7eef409fefbfebfa552ad795f9cb7bff18c8da7588fdd6966cc757161214d348
delete_me
f7b02d2e5ea29180ff52443297a25c86b2897ca30518d5505d5a4e75f6ed17f7
Error response from daemon: Cannot start container f7b02d2e5ea29180ff52443297a25c86b2897ca30518d5505d5a4e75f6ed17f7: a container has already joined the endpoint
```

Some docker logs
```
INFO[0002] DELETE /v1.20/containers/delete_me?force=1   
DEBU[0002] Sending 9 to 7eef409fefbfebfa552ad795f9cb7bff18c8da7588fdd6966cc757161214d348 
ERRO[0002] Leave all failed for  7eef409fefbfebfa552ad795f9cb7bff18c8da7588fdd6966cc757161214d348: could not find sandbox for container id 7eef409fefbfebfa552ad795f9cb7bff18c8da7588fdd6966cc757161214d348 
DEBU[0002] Calling POST /containers/create              
INFO[0002] POST /v1.20/containers/create?name=delete_me 
DEBU[0002] Calling POST /containers/{name:.*}/start     
INFO[0002] POST /v1.20/containers/f7b02d2e5ea29180ff52443297a25c86b2897ca30518d5505d5a4e75f6ed17f7/start 
ERRO[0002] Leave all failed for  f7b02d2e5ea29180ff52443297a25c86b2897ca30518d5505d5a4e75f6ed17f7: could not find sandbox for container id f7b02d2e5ea29180ff52443297a25c86b2897ca30518d5505d5a4e75f6ed17f7 
```

This is broken since #14071 went in.

**Comment by tomdee** [comment]
ping @mavenugo I think this is the issue that was hit during the demo.

**Comment by mavenugo** [comment]
@tomdee thanks. this is consistent for `--net=host`, but i think there are other cases where the problem is seen (such as forceful docker daemon restart).

**Comment by LK4D4** [comment]
Fixed in #14163
